packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -457,6 +457,24 @@ class ThreadChannel extends BaseChannel {
     return this.edit({ appliedTags, reason });
   }
 
+  /**
+   * Pins this thread from the forum channel.
+   * @param {string} [reason] Reason for pinning
+   * @returns {Promise<ThreadChannel>}
+   */
+  pin(reason) {
+    return this.edit({ flags: this.flags.add(ChannelFlags.Pinned), reason });
```

Comment:
```suggestion
    return this.edit({ flags: this.flags.bitfield | ChannelFlags.Pinned, reason });
```

Reply:
this.flags is frozen so it wouldn't be mutated